### PR TITLE
Added Horse Types To Horse.java.

### PR DESCRIPTION
**Pull request by ewlsh**
This doesn't add the 35 different combinations of NORMAL Horses, but it would allow us to set the type of the horse... Source For Types: http://www.minecraftwiki.net/wiki/Item_id#Horse_Variants

### CODE REVIEW

File: src/main/java/org/bukkit/entity/Horse.java [hunk 1]
```diff
@@ -3,4 +3,44 @@
 /**
  * Represents a Horse.
  */
-public interface Horse extends Animals, Vehicle {}
+public interface Horse extends Animals, Vehicle {
+
+    public Type getHorseType();
```
Comment (turt2live) [inline src/main/java/org/bukkit/entity/Horse.java]: Inconsistent return value

Edit: and extra lines below

File: src/main/java/org/bukkit/entity/Horse.java [hunk 2]
```diff
@@ -3,4 +3,44 @@
...
+        public int getId() {
+            return id;
+        }
+
+      
+        public static HorseType getType(int id) {
+            return (id >= types.length) ? null : types[id];
+        }
+    }
+
```
Comment (turt2live) [inline src/main/java/org/bukkit/entity/Horse.java]: More extra lines

File: src/main/java/org/bukkit/entity/Horse.java [hunk 3]
```diff
@@ -3,4 +3,44 @@
...
+        private HorseType(int id) {
+            this.id = id;
+        }
+
+     
+        public int getId() {
+            return id;
+        }
+
```
Comment (turt2live) [inline src/main/java/org/bukkit/entity/Horse.java:52]: Extra lines

File: src/main/java/org/bukkit/entity/Horse.java [hunk 4]
```diff
@@ -3,4 +3,44 @@
...
+        static {
+            for (HorseType type : values()) {
+                types[type.getId()] = type;
+            }
+        }
+
+        private HorseType(int id) {
+            this.id = id;
+        }
+
```
Comment (turt2live) [inline src/main/java/org/bukkit/entity/Horse.java:43]: ~~Extra lines~~

Edit: I'm an idiot.

File: src/main/java/org/bukkit/entity/Horse.java [hunk 5]
```diff
@@ -3,4 +3,44 @@
 /**
  * Represents a Horse.
  */
-public interface Horse extends Animals, Vehicle {}
+public interface Horse extends Animals, Vehicle {
```
Comment (turt2live) [inline src/main/java/org/bukkit/entity/Horse.java:6]: Your newly added methods & enum need JavaDocs

**Comment by ewlsh** [comment]
I think I added what you wanted :P

File: src/main/java/org/bukkit/entity/Horse.java [hunk 6]
```diff
@@ -3,4 +3,66 @@
...
+    public HorseType getHorseType();
+
+     /**
+     * Sets the new type of this horse.
+     *
+     * @param type New type
+     */
+    public void setHorseType(HorseType type);
+
+     /*
```
Comment (turt2live) [inline src/main/java/org/bukkit/entity/Horse.java]: This is a block comment, not JavaDocs

**Comment by turt2live** [comment]
Don't forget the whitespace. A bukkit member will find more than what I have though (At least they usually do)

**Comment by ewlsh** [comment]
Okay... I checked this against Skeleton.java and Ocelot.java and they now mirror each other except for the Horse part, so I don't think anything is wrong anymore.

**Comment by turt2live** [comment]
You still have excess lines

**Comment by ewlsh** [comment]
Good?

**Comment by turt2live** [comment]
Still have a few

**Comment by ewlsh** [comment]
The spacing now matches Skeleton.java ...I found 1 extra space :|

**Comment by turt2live** [comment]
Still one extra line... I have it commented on the diff. 

And don't match to another file in Bukkit, match to the [guidelines](https://github.com/Bukkit/Bukkit/blob/master/CONTRIBUTING.md).

**Comment by ewlsh** [comment]
Can you just give me the line number :|

**Comment by turt2live** [comment]
43, as shown by previous comments.

**Comment by ewlsh** [comment]
Ah yes... I found it, but considering that in the other Bukkit classes that there is a space there... I don't see any reason to remove it :|

**Comment by ewlsh** [comment]
https://github.com/Bukkit/Bukkit/blob/master/src/main/java/org/bukkit/entity/Skeleton.java
- Line 40

**Comment by turt2live** [comment]
Oh, I'm an idiot. I for some reason saw 2 lines there...

Ignore that... 

Looks good, all you have to do now is wait :)

**Comment by ewlsh** [comment]
:P

**Comment by Black-Hole** [comment]
Since this pull request doesn't follow the guidelines for a PR and lacks the implementation part in CraftBukkit, there is no chance this will get pulled.
But I think at some time before the first beta release of 1.6.x Bukkit will come up with a complete solution.
At least you learned some of the steps for a pull request. :)

**Comment by drtshock** [comment]
Follow the contributing guidelines :)

https://github.com/Bukkit/Bukkit/blob/master/CONTRIBUTING.md

File: src/main/java/org/bukkit/entity/Horse.java [hunk 7]
```diff
@@ -3,4 +3,61 @@
...
+        private HorseType(int id) {
+            this.id = id;
+        }
+
+        /**
+         * Gets the ID of this horse type.
+         *
+         * @return Horse type ID
+         */
+        public int getId() {
```
Comment (riking) [inline src/main/java/org/bukkit/entity/Horse.java:49]: I think the mapping of enum -> NMS ID should be done craftbukkit-side.

**Comment by mbax** [comment]
Handled in https://github.com/Bukkit/Bukkit/commit/1a8fb592da70831c3952c7dc8bea9eee0a8527a8

Thanks for the PR :)

**Pull request closed without merging**